src/main/java/io/stargate/sgv2/jsonapi/service/shredding/collections/WritableShreddedDocument.java

```diff
@@ -304,6 +297,28 @@ public void shredVector(JsonPath path, ArrayNode vector) {
       queryVectorValues = arrayVals;
     }
 
+    @Override
+    public void shredVector(JsonPath path, ObjectNode vector) {
+      // vector data is added only to queryVectorValues and exists keys index
+      addKey(path);
+
+      String binaryString = vector.get(JsonExtensionType.BINARY.encodedName()).asText();
+
+      byte[] binaryPayload;
+      binaryPayload = Base64Variants.MIME_NO_LINEFEEDS.decode(binaryString);
+
+      int numFloats = binaryPayload.length / 4;
```

Comment:
As per our comments, should at least verify that byte[] is divisible by 4. So something like:

```
if ((binaryPayload.length & 3) != 0) { 
   // report error
}
```